src/code/circuit_python/gizmo.md

```diff
@@ -0,0 +1,163 @@
+the Gizmo. This includes pin name constants and gamepad values. This is
+all done with the `Gizmo` class.
+
+```Python
+import circuitpython_gizmo
+
+# Create a new Gizmo object
+gizmo = circuitpython_gizmo.Gizmo()
+```
+
+The `Gizmo` class has several useful variables and functions. Firstly, it
+features a number of useful constants for referencing the ports on the
+Gizmo.
+
+- `gizmo.MOTOR_N` - the nth motor port (1 - 8)
+- `gizmo.GPIO_N` - the nth gpio port (1 - 8)
+- `gizmo.ADC_N` - the nth analog port (1 - 3)
+- `gizmo.NEOPIXEL` - the Neopixel port
+- `gizmo.UART_TX` and `gizmo.UART_RX` - the transmit and receive pins for the UART port
+
+The `Gizmo` class can also be used to get the gamepad states.
+
+There are six axes available:
+
+- `gizmo.axes.left_x` - Left stick X-axis (0-255)
+- `gizmo.axes.left_y` - Left stick Y-axis (0-255)
+- `gizmo.axes.right_x` - Right stick X-axis (0-255)
+- `gizmo.axes.right_y` - Right stick Y-axis (0-255)
+- `gizmo.axes.dpad_x` - D-pad X-axis (0, 127, or 255)
+- `gizmo.axes.dpad_y` - D-pad Y-axis (0, 127, or 255)
+
+There are twelve buttons available:
+
+- `gizmo.buttons.x` - X button
+- `gizmo.buttons.a` - A button
+- `gizmo.buttons.b` - B button
+- `gizmo.buttons.y` - Y button
+- `gizmo.buttons.left_shoulder` - Left shoulder button
+- `gizmo.buttons.right_shoulder` - Right shoulder button
+- `gizmo.buttons.left_trigger` - Left trigger
+- `gizmo.buttons.right_trigger` - Right trigger
+- `gizmo.buttons.back` - Back button
+- `gizmo.buttons.start` - Start button
+- `gizmo.buttons.left_stick` - Left joystick stalk
+- `gizmo.buttons.right_stick` - Right joystick stalk
+
+Finally, the `refresh` function must be called at regular intervals to get
+the latest information from the system processor. This is usually done
+towards the start of your program's infinite while loop.
+
+```Python
+while True:
+  gizmo.refresh()
+```
+
+## Logging Gamepad State
+
+Let's put everything we've introduced here together in a simple program
+that counts how many times a button on our gamepad is pressed.
+
+First, import the Circuit Python Gizmo module.
+
+```Python
+import circuitpython_gizmo
+```
+
+Then, create a `Gizmo` object and a counter variable. We will use the
+`count` variable to keep track of how many times the button has been
+pressed.
+
+```Python
+gizmo = circuitpython_gizmo.Gizmo()
+count = 0
+```
+
+Next, in a while loop, refresh the `Gizmo` object to get the latest
+information.
+
+```Python
+while True:
+  gizmo.refresh()
+```
+
+Still in the loop, check if the A button has been pressed. If it has, add
+1 to the count and print it.
+
+```Python
+  if gizmo.buttons.a:
+    count += 1
+    print(f"The button has been pressed {count} times.")
+```
+
+The complete program looks like this:
+
+```Python
+import circuitpython_gizmo
+
+gizmo = circuitpython_gizmo.Gizmo()
+count = 0
+
+while True:
+  gizmo.refresh()
+  if gizmo.buttons.a:
+    count += 1
```

Comment:
I'm pretty sure this increments at a fantastic rate since there's no rising edge detection here.  If that is the case, you might consider adding either the logic for rising edge detection, or changing the description to reveal that this increments while the button is pressed.

Reply:
It sure did. Added the rising edge detection.